fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java

```diff
@@ -1581,11 +1581,9 @@ private PhysicalPartition createPhysicalPartition(String name, Database db, Olap
         }
 
         Long id = GlobalStateMgr.getCurrentState().getNextId();
-        long shardGroupId = 0;
-        if (olapTable.isCloudNativeTableOrMaterializedView()) {
-            shardGroupId = GlobalStateMgr.getCurrentState().getStarOSAgent().
-                    createShardGroup(db.getId(), olapTable.getId(), id);
-        }
+        // physical partitions in the same logical partition use the same shard_group_id,
+        // so that the shards of this logical partition are more evenly distributed.
+        long shardGroupId = partition.getShardGroupId();
```

Comment:
need double check in StarMgrMetaSync, likely can lead to shard deletion by mistake. @starrocks-xupeng please double check on this.